### PR TITLE
[KYUUBI 661] Add UDF system_user

### PR DESCRIPTION
**Pull request by zhaomin1423**
### _Why are the changes needed?_

add udf system_user.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala [hunk 1]
```diff
@@ -36,6 +36,13 @@ object KDFRegistry {
     "string",
     "1.3.0")
 
+  val system_user: KyuubiDefinedFunction = create(
+    "system_user",
+    udf(() => System.getProperty("user.name")).asNonNullable(),
+    "Return the system name for the associated query engine",
```
Comment (turboFei) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala:42]: system user name

File: docs/sql/functions.md [hunk 2]
```diff
@@ -13,4 +13,5 @@ Kyuubi provides several auxiliary SQL functions as supplement to Spark's [Built-
 Name | Description | Return Type | Since
 --- | --- | --- | ---
 kyuubi_version | Return the version of Kyuubi Server | string | 1.3.0
+system_user | Return the system name for the associated query engine | string | 1.3.0
```
Comment (turboFei) [inline docs/sql/functions.md:16]: system user name

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala (hunk 1 above)
Comment (zhaomin1423) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala:42]: Thanks for the suggestion.

**Comment by ulysses-you** [comment]
Just curious, where do you want to use the system user instead of `UserGroupInformation.getCurrentUser` ? For me, it's more like a administrator function. And then maybe we can have some other useful functions `current_machine`, `current_session_number`, `current_statement_number` ?

**Comment by zhaomin1423** [comment]
@ulysses-you  Should the system_user return the UserGroupInformation.getCurrentUser?

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala [hunk 3]
```diff
@@ -36,6 +36,13 @@ object KDFRegistry {
     "string",
     "1.3.0")
 
+  val system_user: KyuubiDefinedFunction = create(
+    "system_user",
+    udf(() => System.getProperty("user.name")).asNonNullable(),
```
Comment (turboFei) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala:41]: should we use a lazy val for `System.getProperty("user.name")` and return it directly? @yaooqinn

**Review by turboFei — APPROVED** [review]
LGTM

**Comment by yaooqinn** [comment]
thanks @zhaomin1423 @turboFei , merged to master

**Pull request closed without merging**